problems/arraysWithLoops.js

```diff
@@ -25,39 +40,86 @@ function shoutWhileLoop() {}
  * @returns {Array} - ["I!", "am!", "a!", "happy!", "array!"]
  */
 
-function shoutForOfLoop() {}
+function shoutForOfLoop(arr) {
+myArray = []
+  for (let i of arr) {
+    myArray.push(`${i}!`);
+    
+  }
+  return myArray;
+    
+  }
+
+
+
+
 
 /**
  * Takes in an array and returns the sum of all values
  * @param {number[]} nums
  * @returns {number} sum
  */
 
-function sumArray() {}
+function sumArray(nums) {
+  let sum = 0;
+
+  for (i = 0; i < nums.length; i++) {
+    console.log(sum);
+    sum += nums[i];
+  }
+  return sum;
+}
 
 /**
  * Takes in an array and returns an array of all the odd valued elements
  * @param {number[]} nums
  * @returns {array} odds
  */
 
-function oddArray() {}
+function oddArray(nums) {
+  let newArray = [];
+
+  for (let i = 0; i < nums.length; i++) {
+    if (nums[i] % 2 === 1) {
+      newArray.push(nums[i]);
+    }
+  }
+  return newArray;
+}
 
 /**
  * Takes in an array and returns an array of all the even valued elements
  * @param {number[]} nums
  * @returns {array} evens
  */
 
-function evenArray() {}
+function evenArray(nums) {
+  let newArray = [];
+
+  for (let i = 0; i < nums.length; i++) {
+    if (nums[i] % 2 === 0) {
+      newArray.push(nums[i]);
+    }
+  }
+  return newArray;
+}
 
 /**
  * Takes in array and returns the smallest number in the array
  * @param {number[]} nums
  * @returns {number} smallest value
  */
 
-function findSmallest() {}
+function findSmallest(nums) {
+  let smallNum = nums[0];
+
+  for (let i = 0; i < nums.length; i++) {
+    if (nums[i] <= smallNum) {
```

Comment:
"<=" works but you can also just do "<" because you won't need to change the variable's value if they're equal